app/src/main/java/org/simple/clinic/settings/SettingsEffectHandler.kt

```diff
@@ -11,20 +11,36 @@ object SettingsEffectHandler {
 
   fun create(
       userSession: UserSession,
+      settingsRepository: SettingsRepository,
       schedulersProvider: SchedulersProvider
   ): ObservableTransformer<SettingsEffect, SettingsEvent> {
     return RxMobius
         .subtypeEffectHandler<SettingsEffect, SettingsEvent>()
         .addTransformer(LoadUserDetailsEffect::class.java, loadUserDetails(userSession, schedulersProvider.io()))
+        .addTransformer(LoadCurrentLanguageEffect::class.java, loadCurrentSelectedLanguage(settingsRepository, schedulersProvider.io()))
         .build()
   }
 
-  private fun loadUserDetails(userSession: UserSession, scheduler: Scheduler): ObservableTransformer<LoadUserDetailsEffect, SettingsEvent> {
-    return ObservableTransformer { upstream ->
-      upstream
+  private fun loadUserDetails(
+      userSession: UserSession,
+      scheduler: Scheduler
+  ): ObservableTransformer<LoadUserDetailsEffect, SettingsEvent> {
+    return ObservableTransformer { effectStream ->
```

Comment:
`loadUserDetailsEffects` instead of `effectStream`?

Reply:
I intentionally called it `effectStream` because I don't want to think about what to name it. The type of effect is anyway *obvious* from the function signature. If there is an effect whose properties I need to consume, I can name it at the site of usage as the lambda parameter.

Does this make sense?